### ISSUE TITLE
Support `NO WAIT` with FOR SHARE on MySQL/Postgres

### ISSUE THREAD

**Issue by petedmarsh** [jOOQ/jOOQ#8054]
At present a `forUpdate` statement does support `noWait()`:

```
select()
        .from(...)
        .forUpdate()
        .noWait();
```

but `forShare()` does not:

```
select()
        .from(...)
        .forShare()
        .noWait(); // No such method!
```

**Comment by lukaseder** [comment]
Thank you very much for your report. We'll look into it soon

**Comment by knutwannheden** [comment]
Looking into this I wonder whether the code in `SelectQueryImpl#setForUpdateOf()`, `SelectQueryImpl#setForUpdateNoWait()`, etc. is correct. These methods all start out by calling `setForUpdate(true)`, which will reset the `forUpdateLockMode` field to `UPDATE`, even though it probably could have had a different value before (e.g. `NO_KEY_UPDATE` or `KEY_SHARE`). I will investigate this further.

**Comment by lukaseder** [comment]
This has been implemented in jOOQ 3.14 via #5218
